File: 2017/test/day_15_test.exs
defmodule AdventOfCode2017.Day15Test do
  use ExUnit.Case

  import AdventOfCode2017.Day15
  import AdventOfCode2017.TestHelper

  test "part1" do
    assert part1(sample_input_1()) == 588
    assert part1(input("test/input/day_15.txt")) == 638
  end

  test "part2" do
    assert part2(sample_input_1()) == 309
    assert part2(input("test/input/day_15.txt")) == 343
  end

  test "parse" do
    assert parse(sample_input_1()) == {65, 8921}
    assert parse(input("test/input/day_15.txt")) == {289, 629}
  end

  defp sample_input_1, do: "Generator A starts with 65\nGenerator B starts with 8921"
end

File: 2018/test/day_15_test.rb
require "test_helper"
require "day_15"

class Day15Test < Minitest::Test
  def setup
    @day_15 = Day15.new
  end

  def test_part1
    assert_equal "5158916779", @day_15.part1(sample_input_1)
  #   assert_equal "0124515891", @day_14.part1(5)
  #   assert_equal "9251071085", @day_14.part1(18)
  #   assert_equal "5941429882", @day_14.part1(2018)
  #   assert_equal "8610321414", @day_14.part1(input_integer("day_14.txt"))
  end

  # def test_part2
  #   assert_equal 9, @day_14.part2("51589")
  #   assert_equal 5, @day_14.part2("01245")
  #   assert_equal 18, @day_14.part2("92510")
  #   assert_equal 2018, @day_14.part2("59414")
  #   assert_equal 20258123, @day_14.part2(input("day_14.txt"))
  # end

  def sample_input_1
    "#########
    #G..G..G#
    #.......#
    #.......#
    #G..E..G#
    #.......#
    #.......#
    #G..G..G#
    #########"
  end
end
